src/tests/ut/reserve_queue/reserve_queue_test.cpp

```diff
@@ -0,0 +1,412 @@
+    sut_.addAwaitingPatient(doctors_[2], patients_[0]);
+    sut_.addAwaitingPatient(doctors_[2], patients_[3]);
+    sut_.addAwaitingPatient(doctors_[2], patients_[2]);
+
+    const auto first_doctor_queued_patients = sut_.awaitingPatients(doctors_[0]);
+    const auto second_doctor_queued_patients = sut_.awaitingPatients(doctors_[1]);
+    const auto third_doctor_queued_patients = sut_.awaitingPatients(doctors_[2]);
+
+    EXPECT_THAT(first_doctor_queued_patients, ElementsAre(patients_[3], patients_[2], patients_[1], patients_[0]));
+    EXPECT_THAT(second_doctor_queued_patients, ElementsAre(patients_[2], patients_[0]));
+    EXPECT_THAT(third_doctor_queued_patients, ElementsAre(patients_[1], patients_[0], patients_[3], patients_[2]));
+}
+
+TEST_F(AddAwaitingPatientShould, NotEnQueuePatientsIfAlreadyEnqueuedForSpecificDoctor)
+{
+    sut_.addAwaitingPatient(doctors_[1], patients_[2]);
+    sut_.addAwaitingPatient(doctors_[1], patients_[1]);
+    sut_.addAwaitingPatient(doctors_[1], patients_[0]);
+    sut_.addAwaitingPatient(doctors_[1], patients_[2]);
+    sut_.addAwaitingPatient(doctors_[1], patients_[0]);
+
+    const auto doctor_queued_patients = sut_.awaitingPatients(doctors_[1]);
+
+    EXPECT_THAT(doctor_queued_patients, ElementsAre(patients_[2], patients_[1], patients_[0]));
+}
+
+class RemoveAwaitingPatientShould : public ReserveQueueFixture
+{
+};
+
+TEST_F(RemoveAwaitingPatientShould, NotChangeTreatmentQueueIfPatientNotPresentInQueue)
+{
+    Treatment filling_treatment{Treatment::DentalFilling};
+    sut_.addAwaitingPatient(filling_treatment, patients_[0]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[1]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[2]);
+
+    Treatment tooth_extraction_treatment{Treatment::ToothExtraction};
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[3]);
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[0]);
+
+    const auto filling_treatment_patients_before = sut_.awaitingPatients(filling_treatment);
+    const auto tooth_extraction_patients_before = sut_.awaitingPatients(tooth_extraction_treatment);
+
+    ASSERT_THAT(filling_treatment_patients_before, ElementsAre(patients_[0], patients_[1], patients_[2]));
+    ASSERT_THAT(tooth_extraction_patients_before, ElementsAre(patients_[3], patients_[0]));
+
+    sut_.removeAwaitingPatient(filling_treatment, patients_[3]);
+    sut_.removeAwaitingPatient(tooth_extraction_treatment, patients_[1]);
+
+    const auto filling_treatment_patients_after = sut_.awaitingPatients(filling_treatment);
+    const auto tooth_extraction_patients_after = sut_.awaitingPatients(tooth_extraction_treatment);
+
+    EXPECT_EQ(filling_treatment_patients_before, filling_treatment_patients_after);
+    EXPECT_EQ(tooth_extraction_patients_before, tooth_extraction_patients_after);
+}
+
+TEST_F(RemoveAwaitingPatientShould, NotChangeDoctorQueueIfPatientNotPresentInQueue)
+{
+    // first doctor queue
```

Comment:
comments not needed, apply to the rest of file

Reply:
DONE. All comments in test file removed.